### PR TITLE
Add E2E test for Relu, Reshape, and Identity op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,7 @@
 *.safetensors
 *.gguf
 *.vmfb
+*.onnx
```
Comment (ScottTodd) [inline .gitignore:31]: I'm not sure about ignoring .onnx files and these other artifact files across the whole repo. Some test cases / suites may benefit from checking in the stable artifact files. Might want to scope those ignores to specific directories.

File: e2eshark/onnx/operators/identity/model.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import numpy, torch, sys
+import onnxruntime
+from onnx import numpy_helper, TensorProto, save_model
+from onnx.helper import make_model, make_node, make_graph, make_tensor_value_info
+
+# import from e2eshark/tools to allow running in current dir, for run through
+# run.pl, commutils is symbolically linked to allow any rundir to work
+sys.path.insert(0, "../../../tools/stubs")
```
Comment (ScottTodd) [inline e2eshark/onnx/operators/identity/model.py:8]: Huh? There should be a more pythonic way to do this. Modifying the system path env var like this is weird.

File: e2eshark/onnx/operators/identity/model.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    {inputs[0].name: model_input_X},
+)
+
+print("Input shape:", model_input_X.shape)
+print("Output shape:", numpy.array(model_output[0]).shape)
+
+# things that need to be kept constant for every test:
+# 1. Define Input and Output Shapes Clearly: Each test must start by defining the input tensor shapes using make_tensor_value_info, ensuring the dimensions match the expected input for the operation being tested.
+# 2. There must be a E2ESHARK_CHECK['input']s and E2ESHARK_CHECK['output']s list so that the test harness code can access the input and output data for validation.
+# 3. The model must be saved as model.onnx in the current working directory
```
Comment (ScottTodd) [inline e2eshark/onnx/operators/identity/model.py:62]: This is a lot of copy/paste boilerplate for an identity model. Can you reduce this file to only what the identity program needs?

Similar comments for the other model.py files.

File: e2eshark/onnx/operators/identity/model.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import numpy, torch, sys
+import onnxruntime
```
Comment (ScottTodd) [inline e2eshark/onnx/operators/identity/model.py:2]: Source files could use license headers (across the entire repo, not just here). Maybe in a separate PR.

Example: https://github.com/nod-ai/SHARK-Turbine/blob/fabd52c50439e59f9ad204f4945eae2cf85a0aa2/core/tests/dynamo/importer_backward_test.py#L1-L7

**Comment by kumardeepakamd** [comment]
Also, at beginning of each model.py you have added, please add:

# Copyright 2024 Advanced Micro Devices
#
# Licensed under the Apache License v2.0 with LLVM Exceptions.
# See https://llvm.org/LICENSE.txt for license information.
# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

File: e2eshark/onnx/operators/reshape/model.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+# Specify the shape of the output
+shape_tensor = make_tensor(
+    name="shape",
+    data_type=TensorProto.INT64,  # Note: Use INT64 for the shape tensor
+    dims=(3,),  # Result tensor will also be of rank 3.
+    vals=[3, 2, 4],
+)
+
+shape_node = make_node(
```
[the remark this answers is not in the document]
Comment (vivekkhandelwal1) [reply e2eshark/onnx/operators/reshape/model.py:33]: It's used here: https://github.com/nod-ai/SHARK-TestSuite/blob/bbb73d301533a208a36a1eca0884b6a863d0caac/e2eshark/onnx/operators/reshape/model.py#L53

**Comment by ScottTodd** [comment]
> Hi @kumardeepakamd @ScottTodd, thanks for the feedback. I have made changes to the PR. Can you guys please review this PR again?

Please click "re-request review" so the PR shows up on my https://github.com/pulls/review-requested page

image

File: .gitignore [hunk 6]
```diff
@@ -28,3 +28,7 @@
 *.safetensors
 *.gguf
 *.vmfb
+e2eshark/onnx/operators/*/model.onnx
```
Comment (ScottTodd) [inline .gitignore:31]: This is fine, but you can also put a .gitignore in the `e2eshark` directory (or any of the other directories). See how I have a similar file here: https://github.com/ScottTodd/SHARK-TestSuite/blob/onnx-testsuite/iree_tests/.gitignore on https://github.com/nod-ai/SHARK-TestSuite/pull/68.

That would be more robust to directory moves. Consider if someone renames `e2eshark` (or moves it up/down a level) they would also need to update the path here in the root file. If there was a .gitignore in that folder itself, the rename would be kept local.

File: e2eshark/onnx/operators/identity/model.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
+# Copyright 2024 Advanced Micro Devices
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (ScottTodd) [inline e2eshark/onnx/operators/identity/model.py:4]: nit: these usually have a line between the copyright and the license
```suggestion
# Copyright 2024 Advanced Micro Devices
#
# Licensed under the Apache License v2.0 with LLVM Exceptions.
# See https://llvm.org/LICENSE.txt for license information.
# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```

File: e2eshark/onnx/operators/relu/model.py
Comment (ScottTodd) [inline e2eshark/onnx/operators/relu/model.py:1]: If you plan on adding more onnx operator tests, let's make sure they are sufficiently different from the test suite I've been hooking up (where ONNX upstream already has done a bulk of the work)

Comment (vivekkhandelwal1) [reply e2eshark/onnx/operators/relu/model.py:1]: Where can I find those tests?

Comment (ScottTodd) [reply e2eshark/onnx/operators/relu/model.py:1]: This PR: https://github.com/nod-ai/SHARK-TestSuite/pull/68

For the relu test:
* Upstream Python definition: https://github.com/onnx/onnx/blob/main/onnx/backend/test/case/node/relu.py
* Upstream generated test case(s): https://github.com/onnx/onnx/tree/main/onnx/backend/test/data/node/test_relu
* New converted test case for IREE: https://github.com/ScottTodd/SHARK-TestSuite/tree/onnx-testsuite/iree_tests/onnx/node/generated/test_relu

These Python tests in SHARK-TestSuite can expose more knobs for changing ONNX/PyTorch/Torch-MLIR/IREE/etc. behavior, but the generated test suites will get us wide coverage that projects like IREE downstream can use directly.

**Review by kumardeepakamd — APPROVED** [review]
Looks good now

**Pull request merged**